### PR TITLE
Enhancements

### LINKED ISSUE #1: Error: The CloudFormation template is invalid: null
Thanks you for providing great plugin! I am following the procedure on README.md but I got this error when I run 'serverless deploy'. Could you help me find what's wrong?

```
$ serverless deploy
Serverless: Packaging service...
Serverless: Using serverless-localstack plugin
Serverless: Using custom endpoint for CloudFormation: http://localstack:4581
Serverless: Using serverless-localstack plugin
Serverless: Using custom endpoint for CloudFormation: http://localstack:4581
Serverless: Uploading CloudFormation file to S3...
Serverless: Using serverless-localstack plugin
Serverless: Using custom endpoint for S3: http://localstack:4572
Serverless: Uploading artifacts...
Serverless: Uploading service .zip file to S3 (2.05 KB)...
Serverless: Using serverless-localstack plugin
Serverless: Using custom endpoint for S3: http://localstack:4572
Serverless: Validating template...
Serverless: Using serverless-localstack plugin
Serverless: Using custom endpoint for CloudFormation: http://localstack:4581
 
  Error --------------------------------------------------
 
  The CloudFormation template is invalid: null
 
     For debugging logs, run again after setting the "SLS_DEBUG=*" environment variable.
 
  Stack Trace --------------------------------------------
 
Error: The CloudFormation template is invalid: null
    at provider.request.catch (/usr/local/lib/node_modules/serverless/lib/plugins/aws/deploy/lib/validateTemplate.js:25:13)
From previous event:
    at PluginManager.invoke (/usr/local/lib/node_modules/serverless/lib/classes/PluginManager.js:217:22)
    at PluginManager.spawn (/usr/local/lib/node_modules/serverless/lib/classes/PluginManager.js:229:17)
    at AwsDeploy.BbPromise.bind.then (/usr/local/lib/node_modules/serverless/lib/plugins/aws/deploy/index.js:88:48)
From previous event:
    at Object.deploy:deploy [as hook] (/usr/local/lib/node_modules/serverless/lib/plugins/aws/deploy/index.js:84:10)
    at BbPromise.reduce (/usr/local/lib/node_modules/serverless/lib/classes/PluginManager.js:217:55)
From previous event:
    at PluginManager.invoke (/usr/local/lib/node_modules/serverless/lib/classes/PluginManager.js:217:22)
    at PluginManager.run (/usr/local/lib/node_modules/serverless/lib/classes/PluginManager.js:236:17)
    at variables.populateService.then (/usr/local/lib/node_modules/serverless/lib/Serverless.js:107:33)
    at runCallback (timers.js:800:20)
    at tryOnImmediate (timers.js:762:5)
    at processImmediate [as _immediateCallback] (timers.js:733:5)
From previous event:
    at Serverless.run (/usr/local/lib/node_modules/serverless/lib/Serverless.js:94:74)
    at serverless.init.then (/usr/local/lib/node_modules/serverless/bin/serverless:30:50)
    at <anonymous>
 
  Get Support --------------------------------------------
     Docs:          docs.serverless.com
     Bugs:          github.com/serverless/serverless/issues
     Forums:        forum.serverless.com
     Chat:          gitter.im/serverless/serverless
 
  Your Environment Information -----------------------------
     OS:                     linux
     Node Version:           8.1.2
     Serverless Version:     1.16.1
 ```

### PR DESCRIPTION
**Pull request by justinm**
Hey temyers,

This PR addresses a few issues I came across while trying to use the plugin. This PR is somewhat of a rewrite but addresses quite a few issues I came across.

Support for configuring endpoints in serverless.yml and JSON files
Rewritten README with improved information on how to use the plugin
Better logging/debugging, added a custom.localstack.debug flag to enable
Verbose logging to intercept AWS HTTP traffic, added a custom.localstack.verbose flag to enable
Preventing infinite loop if Localstack consistently returns a 429 exception
Unit tests passing
Fixes #1

### CODE REVIEW

**Comment by justinm** [comment]
It seems I missed the integration_tests, however the unit tests are passing.

**Comment by adieuadieu** [comment]
@justinm what are your thoughts on #3?

File: README.md [hunk 1]
```diff
@@ -1,36 +1,72 @@
...
 
-Getting Started:
+    npm install -g serverless
+    npm install -g serverless-localstack
 
-* Clone the repository
-`git clone https://github.com/temyers/serverless-localstack`
+## Installation (without npm)
 
-* Start localstack
```
Comment (temyers) [inline README.md:17]: I don't like problems with "it works on my machine", therefore think it's important to include a Docker development environment.  Could you re-instate a section with the info starting the Docker container.  Maybe under a "Development" or "Docker" section.

File: package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
-  "name": "serverless-plugin-localstack",
```
Comment (temyers) [inline package.json:2]: why did you change the name?

**Review by temyers — COMMENTED** [review]
Thanks for the contribution.

There's a couple of changes I've seen so far, but I also want to look into the changes suggested by #3, so that might require a bunch of additional changes following that.

Perhaps we could split this PR into two - one for the doco improvements, and the additional enhancements?

File: README.md (hunk 1 above)
Comment (justinm) [reply README.md:17]: That's a fair point. I think the reason why I decided to pull this section was due to the many different ways to start up Localstack and it's documentation is decent.

File: package.json (hunk 2 above)
Comment (justinm) [reply package.json:2]: Mainly for two reasons:

1) It matches the actual repository name.
2) I took a look at several other plugins listed on the serverless plugins repo and there were a couple of patterns I found: the npm install was the same as the repo name, or repo-name-plugin.

Since this plugin hasn't been published to NPM yet it seems like a good time to propose a name change.

Comment (temyers) [reply package.json:2]: 👍

**Comment by justinm** [comment]
I agree, the proposal in #3 seems like a wonderful direction to move this plugin forward. The idea of hijacking the AWS request method seems dirty.

There are really two features in this PR I was really targetting, configuring endpoints in serverless.yml and the ability to turn on debug/verbose logging as needed. I was attempting to use several features of Localstack that was unsupported or bugged and it was helpful in debugging those issues.

https://bitbucket.org/atlassian/localstack/pull-requests/111/adding-support-for-kinesis-in/diff
https://bitbucket.org/atlassian/localstack/pull-requests/112/46-dynamodb-should-not-enable-streams-by/diff

File: README.md (hunk 1 above)
Comment (temyers) [reply README.md:17]: Thanks.

It also give an environment for contributing changes: create a standardised environment for development and CI.  So maybe it should live under "Contributing"?

**Comment by temyers** [comment]
Awesome that it's been useful even in it's clunky state.

**Comment by temyers** [comment]
If you can update the documentation to include the Docker stuff, then I'll merge this as-is.  The integration tests aren't working on master at the moment, so this is at least a step in the right direction.

**Comment by justinm** [comment]
I've submitted a second PR using the ideas presented in #3. This PR also provides the improvements you requested to the README file.

**Comment by temyers** [comment]
Superceded by #5
